### PR TITLE
Fix Array Knob deserialization

### PR DESCRIPTION
**Pull request by rylnd**
Issue: Array Knob data retrieved from the URL is stored as an Object

## What I did
Updated the Array knob's `deserialize` method to handle Objects.

## How to test
Without these changes, if you reload the page with a component using the `array` knob and expecting an Array; you'll get the error
```
warning.js:33 Warning: Failed prop type: Invalid prop `knob.value` of type `object` supplied to `ArrayType`, expected `array`.
```

### Is this testable with jest or storyshots?
Yes; unit tests have been added for Array's `deserialize`.

### Does this need a new example in the kitchen sink apps?
No.

### Does this need an update to the documentation?
No.

### CODE REVIEW

**Review by danielduan — APPROVED** [review]

**Comment by danielduan** [comment]
Thanks for the PR and the test!

File: addons/knobs/src/components/types/Array.js [hunk 1]
```diff
@@ -47,6 +47,10 @@ ArrayType.propTypes = {
 };
 
 ArrayType.serialize = value => value;
-ArrayType.deserialize = value => value;
+ArrayType.deserialize = value => {
+  if (Array.isArray(value)) return value;
+
+  return Object.keys(value).reduce((array, key) => [...array, value[key]], []);
```
Comment (Hypnosphi) [inline addons/knobs/src/components/types/Array.js]: Does this rely on `Object.keys` being sorted? It's actually not guaranteed to be:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys
> The Object.keys() method returns an array of a given object's own enumerable properties, in the same order as that provided by a for...in loop

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in#Array_iteration_and_for...in
> There is no guarantee that for...in will return the indexes in any particular order

Please sort the keys before reducing

**Review by Hypnosphi — CHANGES_REQUESTED** [review]

File: addons/knobs/src/components/types/Array.js (hunk 1 above)
Comment (rylnd) [reply addons/knobs/src/components/types/Array.js]: @Hypnosphi as evidenced in the unit test, the order of keys does not matter.

Comment (rylnd) [reply addons/knobs/src/components/types/Array.js]: @Hypnosphi 2f22c2c

**Review by Hypnosphi — APPROVED** [review]

**Pull request merged**